lib/src/rules/dartdoc_summary_line.dart

```diff
@@ -0,0 +1,185 @@
+/// [path] or it can't be accessed, this function throws either [IOError]
+/// or [PermissionError], respectively. Otherwise, this deletes the file.
+void delete(String path) {
+  ...
+}
+```
+
+From [Effective Dart](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph):
+
+**DO** separate the first sentence of a doc comment into its own paragraph.
+
+Add a blank line after the first sentence to split it out into its own
+paragraph. If more than a single sentence of explanation is useful, put the
+rest in later paragraphs.
+
+This helps you write a tight first sentence that summarizes the documentation.
+Also, tools like dart doc use the first paragraph as a short summary in places
+like lists of classes and members.
+
+For example:
+
+**GOOD:**
+```dart
+/// Deletes the file at [path].
+///
+/// Throws an [IOError] if the file could not be found. Throws a
+/// [PermissionError] if the file is present but could not be deleted.
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Deletes the file at [path]. Throws an [IOError] if the file could not
+/// be found. Throws a [PermissionError] if the file is present but could
+/// not be deleted.
+void delete(String path) {
+  ...
+}
+```
+''';
+
+/// Enforces that DartDoc blocks start with a single, brief line that ends in
+/// a period.
+///
+/// (Like the one above!)
+class DartdocSummaryLine extends LintRule {
+  DartdocSummaryLine()
+      : super(
+            name: 'dartdoc_summary_line',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addComment(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  /// The summary line should be 'brief' according to the Dart recommendations,
+  /// however 'brief' is not defined perfectly. Instead of enforcing a fixed
+  /// number of characters, we just look at how many lines are part of the
+  /// summary, and if we deem them to be too many (2 should be plenty), we fail
+  /// the lint
+  ///
+  /// The above summary fails with all the checks that this lint adds, it has
+  /// more than one sentence, is too long, and doesn't add a period at the end.
+  ///
+  /// It has it all!
+  @override
+  void visitComment(Comment node) {
+    if (!node.isDocumentation || !node.tokens.first.lexeme.startsWith('///')) {
+      // Ignore /** comments.
+      return;
+    }
+
+    Iterable<Token> summary = _getSummary(node);
+
+    if (!_endsWithPeriod(summary)) {
+      Token last = summary.last;
+      // Highlight the last character of the sentence...
+      rule.reportLintForOffset(last.offset + last.lexeme.length, 1);
+    }
+    int? sentenceBreakOffset = _findSentenceBreak(summary);
```

Comment:
The rule description says that the summary should only be one sentence:

    start doc comments with a single-sentence summary.

and if we just enforced that things would get a lot simpler...

Was that just too limiting in practice?

Reply:
Having more than one sentence is probably myself contradicting my own idea of "just because the markdown parser can handle it, it doesn't mean it's good"; so yeah, maybe just enforcing a literal *one line* as the summary simplifies everything a bunch. Then the length of that line is (maybe) taken care of by another lint rule.